### PR TITLE
Milestone 2: user interface

### PR DESCRIPTION
**Pull request by mahmoud717**
we created a user interface that has an infrastructure of the game using comments and also does the following:
1. register both players' names and tokens.
2. askes the players for a line and a column and registers them in separate values for each player.
3. display the chosen play after the player selects it.

### CODE REVIEW

**Comment by santiago-dev96** [comment]
# Pull Request under review by @santiago-rodrig (TSE)

File: bin/main [hunk 1]
```diff
@@ -1,2 +1,58 @@
 #!/usr/bin/env ruby
-puts 'Hello, world! '
+
+# get the player1 name
+p 'Hey player1, whats your name?'
```
Comment (santiago-dev96) [inline bin/main]:
```suggestion
puts "Hey player 1, what's your name?"
```

File: bin/main [hunk 2]
```diff
@@ -1,2 +1,58 @@
...
+player1 = gets.chomp
+p "hello #{player1}, Please select a token (x / o)"
+# get the player1 token (x / o)
+player1_token = gets.chomp
+p "#{player1_token}, Nice choice!"
+
+# get the player2 name
+p 'Hey player2, whats your name?'
```
Comment (santiago-dev96) [inline bin/main]:
```suggestion
puts "Hey player 2, what's your name?"
```

File: bin/main [hunk 3]
```diff
@@ -1,2 +1,58 @@
...
+player2 = gets.chomp
+# get the player2 token (x / o)
+player2_token = if player1_token == 'x'
+                  'o'
+                else 'x'
+                end
```
Comment (santiago-dev96) [inline bin/main]:
```suggestion
player2_token = player1_token == 'x' ? 'o' : 'x'
```

File: bin/main [hunk 4]
```diff
@@ -1,2 +1,58 @@
...
+p "hello #{player2}, your token is #{player2_token}"
```
Comment (santiago-dev96) [inline bin/main]:
```suggestion
puts "hello #{player2}, your token is #{player2_token}"
```

File: bin/main [hunk 5]
```diff
@@ -1,2 +1,58 @@
...
+
+# create a loop to take each player move
+5.times do
```
Comment (santiago-dev96) [inline bin/main]: This is not OK, you should have a `game_over` boolean variable that tells when the game is over and iterate until this variable is `true`.

File: bin/main [hunk 6]
```diff
@@ -1,2 +1,58 @@
...
+  # check if the column is valid
+
+  # confirm selected move
+  puts "You selected line #{player2_line} and column #{player2_column}."
+  # display the board
+  # check if the game is complete
+  # return a winner
+  puts "Hurray, #{player2} you win"
+  # else
+  # next
```
Comment (santiago-dev96) [inline bin/main]: This is not a good approach to mocking the game flow, first, the positions in the board should be plain integers, starting at 1 and ending at 9.

1 | 2 | 3
4 | 5 | 6
7 | 8 | 9

That's the expected aspect of the board, of course for this milestone it is not required to change the state of the board in any form, so you can just display it after every player's turn.

You can mock all the possible scenarios, X wins, O wins, and draw, but make sure you have some sort of counter and that the scenario is selected at random (`rand(3)` can help), think about that by yourselves.

**Review by santiago-dev96 — CHANGES_REQUESTED** [review]
# Changes required 🛠️ 

This is a good start, but some things need to be fixed, please read the comments I left there to help, the goal of this project is to fake the game flow, that doesn't mean using `5.times`, because that doesn't take into account the draw state of the game, or how the game looks when X wins or O wins, you must think on how the user will perceive those parts of the game.

Another thing, don't make the user input row and column when just passing a number will suffice, and print the board after every player's turn.

### [OPTIONAL]

You have some typos in your `readme.md`, and this file should have a name with just uppercase letters, like `README.md`, apart from that, congratulations for including a `README.md` file even though it is not required yet for this milestone.

## What to do now?

Implement the required changes and submit another CRR.

You can find me in slack as **srodrig**, happy coding! :wave:

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hi Team 👋

Nice work so far on this. 👌👌


- [ ] Sorry this wasn't mentioned in the least review but please add a rubocop.yml file as [mentioned here](https://github.com/microverseinc/linters-config/tree/master/ruby).

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

File: bin/main [hunk 7]
```diff
@@ -1,2 +1,55 @@
...
+  puts "#{player2}, choose a position"
+  player2_position = gets.chomp
+  # check if the position is valid
+  # confirm selected move
+  puts "You selected position #{player2_position}."
+  # display the board
+  display_board
+  # check if the game is complete
+  # return a winner
+  puts "Hurray, #{player2} you win"
```
Comment (mikenath223) [inline bin/main]: - [ ] Notice that you are repeating the same process for each player's turn. Consider extracting this as a method you would call in the loop for each player turn.

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hi again Team. 👋

Good job fixing an issue referenced in the previous review.👍

However, kindly take note of the comments below as this was referenced in the last review. It's important you follow the DRY principle.👇

Kindly fix this issue and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

File: bin/main (hunk 7 above)
Comment (mikenath223) [reply bin/main:52]: > * [ ]  Notice that you are repeating the same process for each player's turn. Consider extracting this as a method you would call in the loop for each player turn.

Please consider extracting the repeated code in a method like this and calling it in each player's turn within the loop:
```
def each_turn(player)
  puts "#{player}, choose a position"
  player_position = gets.chomp
  # check if the position is valid
  # confirm selected move
  puts "You selected position #{player_position}."
  # display the board
  display_board
end
```

File: bin/main [hunk 8]
```diff
@@ -1,2 +1,56 @@
...
+  # else
+  # next
+  # get the player2 move
+  puts "#{player2}, choose a position"
+  player2_position = gets.chomp
+  # check if the position is valid
+  # confirm selected move
+  puts "You selected position #{player2_position}."
+  # display the board
+  display_board
```
Comment (mikenath223) [inline bin/main]: > * [ ]  Notice that you are repeating the same process for each player's turn. Consider extracting this as a method you would call in the loop for each player turn.

Please consider extracting the repeated code in a method like this and calling it in each player's turn within the loop:
```
def each_turn(player)
  puts "#{player}, choose a position"
  player_position = gets.chomp
  # check if the position is valid
  # confirm selected move
  puts "You selected position #{player_position}."
  # display the board
  display_board
end
```

**Comment by Dandush03** [comment]
# Status: :rocket: Project Approved :rocket: 
## Great job applying does changes
# Happy Coding! :computer: :nerd_face:

**Pull request merged**